### PR TITLE
Modify EVMs Transactions to Handle for L2s

### PR DESCRIPTION
**Pull request by MSilb7**
Reverting changes in #3579 to handle for special fields in L2s (and other chains) without creating duplicates issues.

Also creates a macro for `all_op_chains` that can be used to handle for this custom logic, since OP Chains should all have identical schema in Dune.

cc @hildobby 

EDIT: This should come after https://github.com/duneanalytics/spellbook/pull/3640

# SPELLBOOK FREEZE

From June 22 to June 29, we will be freezing some Spellbook contributions for the migration to DuneSQL. We will not accept contributions to the lineage of the following models:

* dex.trades
* nft.trades
* labels
* token.erc20
* tokens.nft

Run the following command to see the list of affected files:

```
dbt ls --resource-type model --output path --select +dex_trades +labels +nft_trades +tokens_nft +tokens_erc20
```

Don't hesitate to reach out on Discord if you have any questions.

Brief comments on the purpose of your changes:

**For Dune Engine V2**

I've checked that:

### General checks:
* [ ] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [ ] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [ ] if adding a new model, I added a test
* [ ] the filename is unique and ends with .sql
* [ ] each sql file is a select statement and has only one view, table or function defined  
* [ ] column names are `lowercase_snake_cased`
* [ ] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [ ] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Pricing checks:
* [ ] `coin_id` represents the ID of the coin on coinpaprika.com
* [ ] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [ ] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [ ] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [ ] where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

**Comment by MSilb7** [comment]
^ aside from that comment, this is good @jeff-dude 

Main purpose is add Goerli (and any future testnets) to evms in a way that doesn't confuse people who may use evms to mean "mainnets" (per @hildobby's recommendation).

I use mainnets + testnets together for developer metrics (see: https://dune.com/queries/2382060/3906387)

**Comment by jeff-dude** [comment]
> EDIT: This should come after #3640 & #3653

based on this, i'm waiting here

**Comment by MSilb7** [comment]
> > EDIT: This should come after #3640 & #3653
> 
> based on this, i'm waiting here

Actually, this shouldn't depend on #3653, editing the first message. It should be good to go @jeff-dude

File: models/evms/evms_transactions.sql [hunk 1]
```diff
@@ -44,49 +46,41 @@ FROM (
...
+                , l1_fee_scalar
+                , l1_block_number
+                , l1_fee
+                , l1_gas_price
+                , l1_gas_used
+                , l1_timestamp
+                , CAST(NULL AS DECIMAL(38,0)) AS effective_gas_price
+
+                {% elif transactions_model[0] == 'arbitrum' %}
+                , NULL AS l1_tx_origin
```
[the remark this answers is not in the document]
Comment (MSilb7) [reply models/evms/evms_transactions.sql]: changing

**Review by jeff-dude — APPROVED** [review]
lgtm

**Comment by jeff-dude** [comment]
blocked until CLA signed

**Pull request merged**